homeassistant/components/intellifire/config_flow.py

```diff
@@ -31,258 +49,217 @@ class DiscoveredHostInfo:
-        """Validate local config and continue."""
-        self._async_abort_entries_match({CONF_HOST: host})
-        self._serial = await validate_host_input(host)
-        await self.async_set_unique_id(self._serial, raise_on_progress=False)
-        self._abort_if_unique_id_configured(updates={CONF_HOST: host})
-        # Store current data and jump to next stage
-        self._host = host
-
-        return await self.async_step_api_config()
-
-    async def async_step_manual_device_entry(self, user_input=None):
-        """Handle manual input of local IP configuration."""
-        LOGGER.debug("STEP: manual_device_entry")
-        errors = {}
-        self._host = user_input.get(CONF_HOST) if user_input else None
         if user_input is not None:
             try:
-                return await self._async_validate_ip_and_continue(self._host)
-            except (ConnectionError, ClientConnectionError):
-                errors["base"] = "cannot_connect"
+                async with self.cloud_api_interface as cloud_interface:
+                    await cloud_interface.login_with_credentials(
+                        username=user_input[CONF_USERNAME],
+                        password=user_input[CONF_PASSWORD],
+                    )
+
+                # Mark as logged in
+                self.logged_in = True
+
+                # If login was successful pass username/password to next step
+                return await self.async_step_pick_cloud_device()
+            except LoginError:
+                errors["base"] = "api_error"
 
         return self.async_show_form(
-            step_id="manual_device_entry",
+            step_id="cloud_api",
             errors=errors,
-            data_schema=vol.Schema({vol.Required(CONF_HOST, default=self._host): str}),
+            data_schema=vol.Schema(
+                {
+                    vol.Required(CONF_USERNAME): str,
+                    vol.Required(CONF_PASSWORD): str,
+                }
+            ),
         )
 
-    async def async_step_pick_device(
+    async def async_step_pick_cloud_device(
         self, user_input: dict[str, Any] | None = None
     ) -> ConfigFlowResult:
-        """Pick which device to configure."""
-        errors = {}
-        LOGGER.debug("STEP: pick_device")
+        """Step to select a device from the cloud.
+
+        We can only get here if we have logged in. If there is only one device available it will be auto-configured,
+        else the user will be given a choice to pick a device.
+        """
+        errors: dict[str, str] = {}
+        LOGGER.debug(
+            f"STEP: pick_cloud_device: {user_input} - DHCP_MODE[{self._dhcp_mode}"
+        )
 
-        if user_input is not None:
-            if user_input[CONF_HOST] == MANUAL_ENTRY_STRING:
-                return await self.async_step_manual_device_entry()
+        if self._dhcp_mode or user_input is not None:
+            if self._dhcp_mode:
```

Comment:
The dhcp_mode field feels strange. Haven't seen that before

Reply:
Only since March 2022 or so:

https://github.com/home-assistant/core/pull/67053

<img width="818" alt="image" src="https://github.com/user-attachments/assets/fc3916a5-027e-4849-af1f-178aa32398b9">